Framework/include/QualityControl/runnerUtils.h

```diff
@@ -74,87 +74,44 @@ inline bool hasChecks(std::string configSource)
-    ILOG(Debug, Devel) << "Got this property runType from RawDeviceService: '" << temp << "'" << ENDM;
-    runType = stoi(temp);
-    ILOG(Debug, Devel) << "   Run type found in options: " << runType << ENDM;
+    auto temp = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>(name, "unspecified");
+    ILOG(Debug, Devel) << "Got this property " << name << " from RawDeviceService: '" << temp << "'" << ENDM;
+    result = boost::lexical_cast<T>(temp);
   } catch (std::invalid_argument& ia) {
-    ILOG(Debug, Devel) << "   Run type not found in options or is not a number, using the one from the config file or 0 as a last resort."
-                       << ENDM;
+    ILOG(Debug, Devel) << "   " << name << " not found in options or is not a number, using the fallback." << ENDM;
   }
-  runType = runType > 0 /* found it in service */ ? runType : fallbackRunType;
-  ILOG(Info, Devel) << "Run type returned by computeRunType (default) : " << runType << ENDM;
-  return runType;
-}
-
-inline std::string computePartitionName(framework::ServiceRegistryRef services, const std::string& fallbackPartitionName = "")
-{
-  std::string partitionName;
-  partitionName = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("environment_id", "unspecified");
-  ILOG(Debug, Devel) << "Got this property partitionName from RawDeviceService: '" << partitionName << "'" << ENDM;
-  partitionName = partitionName != "unspecified" /* found it in service */ ? partitionName : fallbackPartitionName;
-  ILOG(Info, Devel) << "Partition Name returned by computePartitionName : " << partitionName << ENDM;
-  return partitionName;
-}
-
-inline std::string computePeriodName(framework::ServiceRegistryRef services, const std::string& fallbackPeriodName = "")
-{ // Determine period
-  std::string periodName;
-  periodName = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("periodName", "unspecified");
-  ILOG(Debug, Devel) << "Got this property periodName from RawDeviceService: '" << periodName << "'" << ENDM;
-  periodName = periodName != "unspecified" /* found it in service */ ? periodName : fallbackPeriodName;
-  ILOG(Info, Devel) << "Period Name returned by computePeriodName : " << periodName << ENDM;
-  return periodName;
-}
-
-inline std::string computePassName(const std::string& fallbackPassName = "")
-{
-  ILOG(Debug, Devel) << "Pass Name returned by computePassName : " << fallbackPassName << ENDM;
-  return fallbackPassName;
-}
-
-inline std::string computeProvenance(const std::string& fallbackProvenance = "")
-{
-  ILOG(Debug, Devel) << "Provenance returned by computeProvenance : " << fallbackProvenance << ENDM;
-  return fallbackProvenance;
+  result = result > 0 /* found it in service */ ? result : fallbackNumber;
+  ILOG(Info, Devel) << name << " returned by computeRunNumber (default) : " << result << ENDM;
```

Comment:
this log could be updated if you use this template not only for run number anymore.

Reply:
correct, I'll fix that